source-container-build/app/source_build.py

```diff
@@ -710,35 +710,43 @@ def index(self) -> Index:
         return self._index
 
 
-def merge_image(from_image: str, to_image: str) -> None:
-    """Merge sources from one to another
+def merge_image(parent_sources_dir: str, local_source_build: str) -> None:
+    """Merge parent sources into the local source build
 
-    Image layers are appended to the destination image one by one.
+    Layers and associated data are prepended into the local source build and
+    kept the same order as in the parent source container.
 
-    :param from_image: str, merge sources from this image to another.
-    :param to_image: str, sources are merged into this image. Both of
-        ``from_image`` and ``to_image`` are directory paths holding sources in
-        OCI image layout format.
+    :param parent_sources_dir: str, merge sources from this image to another.
+    :param local_source_build: str, sources are merged into this image. Both of
+        ``parent_sources_dir`` and ``local_source_build`` are directory paths
+        holding sources in OCI image layout format.
     """
 
-    parent_image = OCIImage(from_image)
-    local_build = OCIImage(to_image)
+    parent_image = OCIImage(parent_sources_dir)
+    local_build = OCIImage(local_source_build)
 
     parent_image_manifest = parent_image.index.manifests()[0]
     local_build_manifest = local_build.index.manifests()[0]
-    for layer in parent_image_manifest.layers:
+    for layer in reversed(parent_image_manifest.layers):
         copy_dest_path = Path(local_build.path, "blobs", *layer.descriptor["digest"].split(":"))
         shutil.copyfile(layer.path, copy_dest_path)
         logger.debug("copy layer %s to %s", layer.path, copy_dest_path)
-        local_build_manifest.append_layer(layer)
+        local_build_manifest.prepend_layer(layer)
```

Comment:
Maybe it would be simpler to do things like this?

```python
for layer in local_build_manifest.layers:
    ...
    parent_image_manifest.append_layer(layer)

...
parent_image_manifest.save()
```

But that's a nitpick at this point, the current solution seems to work